### PR TITLE
parallel: add disableNagMessage

### PR DESCRIPTION
**Pull request by tomberek**
## Motivation for this change
Concern about the nag screen as well as GPL issues.

See https://github.com/NixOS/nixpkgs/issues/110584
Let's break the problem down into it's components.

###### Nag
This is a usability concern for some people and for some automation.

###### GPLv3
I don't think Nixpkgs should get into the habit of "fixing" the potential license issue by patching this out completely. My personal opinion is that it is a bit non-standard, but is technically okay. Regardless, the Nix-community is not a compliance organization, nor is in a good position to enforce this. The author and the GNU organization seems to claim there is no conflict. I recommend this be brought up with them directly. Packaging systems hiding the potential issue from users by default also seems odd.

###### Things done
adds an option for Nix users to disable the citation request by adding a `--will-cite` to the wrapped parallel call.

###### Potential issues
- some users may want this without the extra wrapping (this would mean a patch)
- some may want to disable the nag, but also not use the extra perl pacakges (this would mean they can set extraPerlPackages to `[]`)

###### Checklist
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tomberek** [comment]
`disableNagMessage` -> `willCite`? Calling it a nag has a negative connotation.

File: pkgs/tools/misc/parallel/wrapper.nix [hunk 1]
```diff
@@ -2,12 +2,15 @@
 , perlPackages
 , extraPerlPackages ?
     with perlPackages; [ DBI DBDPg DBDSQLite DBDCSV TextCSV ]
+, disableNagMessage ? false
+, lib
```
Comment (cole-h) [inline pkgs/tools/misc/parallel/wrapper.nix:6]: Nit (sorry): `lib` should be the first input (as noted by https://github.com/nixos/nixpkgs/blob/1ce696fb1d0ce011a19bdfdc495f93e442d23feb/doc/contributing/coding-conventions.xml#L183-L184).

**Comment by Profpatsch** [comment]
:+1: this looks like a good low-overhead solution, since there is already a wrapper anyway.

**Comment by Profpatsch** [comment]
only thing I would test whether `parallel` is okay with getting passed `--will-cite` twice, in order to not break existing calls that include it. We might have to deduplicate it.

**Comment by cole-h** [comment]
I don't know how to use `parallel`, but running `echo "echo asdf; echo fdsa" | parallel --will-cite --will-cite` doesn't make it angry.

**Comment by ole-tange** [comment]
It seems there is some confusion as to:

* whether the citation notice breaks scripts (It does not. People wrongly claim so but not a single time has that been shown to happen. https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt#n38)
* whether the notice is GPL3 compliant (It is according to RMS https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt#n27)
* whether you can patch out the notice and thus fork the program (You can, but then you cannot call the program GNU Parallel nor can use "GNU Parallel" in the description or documentation: https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt#n110).
* whether it is preferable to not be distributed at all (or distributed as non-free) over not having the notice (It is preferable not to be distributed at all https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt#n133)

Please read: https://git.savannah.gnu.org/cgit/parallel.git/tree/doc/citation-notice-faq.txt

Especially the last section is relevant.

**Comment by tomberek** [comment]
@ole-tange Is the mechanism in this PR acceptable? It is false by default and requires positive action by a user to remove the notice, similar to someone setting their own alias or running `--citation` themselves. (perhaps a bit more documentation needed for the option? perhaps showing the citation in a Nix trace?)

**Comment by ole-tange** [comment]
> @ole-tange Is the mechanism in this PR acceptable?

If the user by default is presented with the citation notice, it is fine.

The fundamental issue is to make sure the users are aware, that citations are what (indirectly) funds future development. If this can be accomplished in a more effective way, it is OK.

**Comment by Profpatsch** [comment]
> If the user by default is presented with the citation notice, it is fine.

So I guess this goes against disabling the nag message by default, per the author’s (@ole-tange) request.

**Comment by tomberek** [comment]
> > If the user by default is presented with the citation notice, it is fine.
> 
> So I guess this goes against disabling the nag message by default, per the author’s (@ole-tange) request.

My impression is that this patch is in accordance to the author's request. By default, the package will present the citation message. Only by specifically installing parallel-full and overriding the default `willCite` option will the message be silenced. This is a similar level of action as a script to run `touch ~/.parallel/will-cite` or to alias parallel. This would only be done by someone who is already familiar with the requests and understands what that option means. It is also similar to our policy with config.allowUnfree or CUDA support in various packages.

I also want to note that I do not particularly believe this proposal is necessary [I am starting to think it would be convenient in certain automation scenarios], but it is a compromise between various positions.

**Pull request merged**